src/Components/Inputs/AutoInput.tsx

```diff
@@ -0,0 +1,102 @@
+// Libraries
+import React, {Component} from 'react'
+import classnames from 'classnames'
+
+// Components
+import {Radio} from '../Radio/Radio'
+
+// Types
+import {
+  StandardProps,
+  ButtonShape,
+  ComponentColor,
+  ComponentSize,
+  AutoInputMode,
+} from '../../Types'
+
+// Styles
+import './AutoInput.scss'
+
+interface Props extends StandardProps {
+  /** Pass in a component of type "Input" */
+  inputComponent: JSX.Element
+  /** Fires when the radio is toggled and the mode changes */
+  onModeChange: (mode: AutoInputMode) => void
```

Comment:
(Minor thing) I think 
```suggestion
  onChangeMode: (mode: AutoInputMode) => void
```
would be more similar to the other components / more predictable to use.

Reply:
Done!